pkg/api/errors/errors.go

```diff
@@ -218,34 +223,42 @@ func NewServiceUnavailable(reason string) error {
 }
 
 // NewMethodNotSupported returns an error indicating the requested action is not supported on this kind.
-func NewMethodNotSupported(kind, action string) error {
+func NewMethodNotSupported(qualifiedResource unversioned.GroupResource, action string) error {
```

Comment:
@smarterclayton methods are associated with resources, not with kinds.  This is particular important since a kind could be used for multiple resources and so the kind is a perfectly valid thing to submit to a different resource with this method.